### PR TITLE
NEWTAG-483 - Require PaperInfo to have a SORTKEY

### PR DESCRIPTION
**Pull request by thpr**

### CODE REVIEW

**Comment by LegacyKing** [comment]
Looks great!

File: code/src/java/plugin/lsttokens/paper/SortKeyToken.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+import pcgen.core.PaperInfo;
+import pcgen.rules.context.LoadContext;
+import pcgen.rules.persistence.token.AbstractNonEmptyToken;
+import pcgen.rules.persistence.token.CDOMPrimaryToken;
+import pcgen.rules.persistence.token.ParseResult;
+import pcgen.rules.persistence.token.PostValidationToken;
+import pcgen.util.Logging;
+
+public class SortKeyToken extends AbstractNonEmptyToken<PaperInfo> implements
```
Comment (jdempsey-au) [inline code/src/java/plugin/lsttokens/paper/SortKeyToken.java]: Class javadoc here would be useful

File: system/gameModes/default/paperInfo.lst [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
+NAME:A4		SORTKEY:001	HEIGHT:297mm	WIDTH:210mm	TOPMARGIN:10mm		BOTTOMMARGIN:15mm	LEFTMARGIN:10mm		RIGHTMARGIN:10mm
+NAME:A5		SORTKEY:002	HEIGHT:148mm	WIDTH:210mm	TOPMARGIN:10mm		BOTTOMMARGIN:15mm	LEFTMARGIN:10mm		RIGHTMARGIN:10mm
+NAME:B5		SORTKEY:003	HEIGHT:257mm	WIDTH:182mm	TOPMARGIN:10mm		BOTTOMMARGIN:15mm	LEFTMARGIN:10mm		RIGHTMARGIN:10mm
 #
-NAME:in_PaperExecutive	HEIGHT:10.5in	WIDTH:7.25in	TOPMARGIN:.4in		BOTTOMMARGIN:.6in	LEFTMARGIN:.4in		RIGHTMARGIN:.4in
-NAME:in_PaperLegal	HEIGHT:14in	WIDTH:8.5in	TOPMARGIN:.4in		BOTTOMMARGIN:.6in	LEFTMARGIN:.4in		RIGHTMARGIN:.4in
-NAME:in_PaperLetter	HEIGHT:11in	WIDTH:8.5in	TOPMARGIN:.4in		BOTTOMMARGIN:.6in	LEFTMARGIN:.4in		RIGHTMARGIN:.4in
+NAME:in_PaperExecutive	SORTKEY:004	HEIGHT:10.5in	WIDTH:7.25in	TOPMARGIN:.4in		BOTTOMMARGIN:.6in	LEFTMARGIN:.4in		RIGHTMARGIN:.4in
+NAME:in_PaperLegal		SORTKEY:005	HEIGHT:14in	WIDTH:8.5in	TOPMARGIN:.4in		BOTTOMMARGIN:.6in	LEFTMARGIN:.4in		RIGHTMARGIN:.4in
+NAME:in_PaperLetter		SORTKEY:006	HEIGHT:11in	WIDTH:8.5in	TOPMARGIN:.4in		BOTTOMMARGIN:.6in	LEFTMARGIN:.4in		RIGHTMARGIN:.4in
```
Comment (jdempsey-au) [inline system/gameModes/default/paperInfo.lst]: Should we order these by alpha or by most used?

**Comment by thpr** [comment]
Added Javadoc per comment.

On ordering, I have left the order as the original file... and for now the SORTKEY has to match that until we enable it to be fully sorted using that value.

**Pull request merged**